core/payments/reservation/reservation_vault_monitor.go

```diff
@@ -101,6 +97,54 @@ func (vm *ReservationVaultMonitor) refreshReservations(ctx context.Context) erro
 	return nil
 }
 
+// Fetches reservations from the PaymentVault. If number of accountIDs exceeds configured rpcBatchSize, multiple RPC
+// calls will be made in series to fetch all reservation data. If rpcBatchSize is configured to be 0, all data
+// will be fetched in a single call, no matter how many accounts are passed in.
+func (vm *ReservationVaultMonitor) fetchReservations(
+	ctx context.Context,
+	accountIDs []gethcommon.Address,
+) ([]*bindings.IPaymentVaultReservation, error) {
+	if len(accountIDs) == 0 {
+		return nil, nil
+	}
```

Comment:
is this really what we want? or is this a bug?

Reply:
You're right, this shouldn't be here. It's being checked that `accountIDs != 0` in `refreshReservations`, prior to calling `fetchReservations`. 

I removed it
cd82cc02a50bedbbcf034e02368fd080fa95c059

Reply:
> But perhaps if 0 len accountIDs is passed itll just work fine anyways since the for loop wont do anything?

That's right. If empty accounts _did_ manage to get passed in, the whole thing would just be a noop